File: src/com/pytest_support/fixtures/FixtureUsageTargetProvider.java
package com.pytest_support.fixtures;

import com.intellij.find.findUsages.FindUsagesHandler;
import com.intellij.find.findUsages.FindUsagesHandlerFactory;
import com.intellij.psi.PsiElement;
import com.jetbrains.python.psi.PyFunction;
import com.pytest_support.utils.FixtureUtils;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Objects;

public class FixtureUsageTargetProvider extends FindUsagesHandlerFactory {
    @Override
    public boolean canFindUsages(@NotNull PsiElement psiElement) {
        if (psiElement instanceof PyFunction) {
            return FixtureUtils.isFixture((PyFunction) psiElement);
        }
        return false;
    }


    @Nullable
    @Override
    public FindUsagesHandler createFindUsagesHandler(@NotNull PsiElement psiElement, boolean b) {
        return new FindUsagesHandler(psiElement) {
            @NotNull
            @Override
            public PsiElement[] getPrimaryElements() {
                String fixtureName = ((PyFunction) getPsiElement()).getName();
                if (fixtureName == null) {
                    return new PsiElement[]{};
                }

                return FixtureUtils
                        .pyFileTopLevelFunctionsAndClassMethods(getProject())
                        .filter(FixtureUtils::isTestFunctionOrFixture)
                        .map(function -> function.getParameterList().findParameterByName(fixtureName))
                        .filter(Objects::nonNull)
                        .distinct()
                        .toArray(PsiElement[]::new);
            }
        };
    }


}

File: src/com/pytest_support/fixtures/FixtureUsageTargetProvider.kt
package com.pytest_support.fixtures

import com.intellij.find.findUsages.FindUsagesHandler
import com.intellij.find.findUsages.FindUsagesHandlerFactory
import com.intellij.psi.PsiElement
import com.jetbrains.python.psi.PyFunction
import com.pytest_support.extensions.allFixtureAndTests
import com.pytest_support.extensions.isFixture
import com.pytest_support.extensions.toParameter
import com.pytest_support.extensions.toPsiArray

class FixtureUsageTargetProvider : FindUsagesHandlerFactory() {
    override fun canFindUsages(psiElement: PsiElement): Boolean =
            if (psiElement is PyFunction) psiElement.isFixture else false

    override fun createFindUsagesHandler(psiElement: PsiElement, b: Boolean): FindUsagesHandler? {
        return object : FindUsagesHandler(psiElement) {
            override fun getPrimaryElements(): Array<PsiElement> {
                val fixtureName = (getPsiElement() as PyFunction).name ?: return emptyArray()

                return project
                        .allFixtureAndTests()
                        .toParameter(fixtureName)
                        .distinct()
                        .toPsiArray()
            }
        }
    }


}
